R/interop-car.R

```diff
@@ -97,41 +97,335 @@ h_get_contrast <- function(object, effect, type = c("II", "III", "2", "3"), tol
-Anova.mmrm <- function(mod, type = c("II", "III", "2", "3"), tol = sqrt(.Machine$double.eps), ...) { # nolint
-  assert_double(tol, finite = TRUE, len = 1L)
+#'
+#' @examples
+#'
+#' \dontshow{
+#'
+#' Please do not load `car` and then create the documentation. The Rd file will
+#' be different.}
+Anova.mmrm <- function(mod,
+                       type = c("II", "III", "2", "3"),
+                       tol = sqrt(.Machine$double.eps),
+                       test.statistic = c("F", "Chisq"),
+                       vcov. = vcov(mod, complete = FALSE),
+                       singular.ok = FALSE,
+                       ...) { # nolint
+  assert_class(mod, "mmrm")
+  type <- as.character(type)
   type <- match.arg(type)
-  vars <- colnames(attr(terms(mod$formula_parts$model_formula), "factors"))
-  ret <- lapply(
-    vars,
-    function(x) df_md(mod, h_get_contrast(mod, x, type, tol))
-  )
-  ret_df <- do.call(rbind.data.frame, ret)
-  row.names(ret_df) <- vars
-  colnames(ret_df) <- c("Num Df", "Denom Df", "F Statistic", "Pr(>=F)")
-  class(ret_df) <- c("anova", "data.frame")
-  attr(ret_df, "heading") <- sprintf(
-    "Analysis of Fixed Effect Table (Type %s F tests)",
-    switch(type,
-      "2" = ,
-      "II" = "II",
-      "3" = ,
-      "III" = "III"
+  assert_double(tol, finite = TRUE, len = 1L)
+  test.statistic <- match.arg(test.statistic)
+
+  if (test.statistic == "F") {
+    vars <- colnames(attr(terms(mod$formula_parts$model_formula), "factors"))
+    ret <- lapply(
+      vars,
+      function(x) df_md(mod, h_get_contrast(mod, x, type, tol))
     )
-  )
+    ret_df <- do.call(rbind.data.frame, ret)
+    row.names(ret_df) <- vars
+    colnames(ret_df) <- c("Num Df", "Denom Df", "F Statistic", "Pr(>=F)")
+    attr(ret_df, "heading") <-
+      sprintf(
+        "Analysis of Fixed Effect Table (Type %s %s tests)",
+        switch(type, "2" = , "II" = "II", "3" = , "III" = "III"),
+        test.statistic
+      )
+    class(ret_df) <- c("anova", "data.frame")
+  } else {
+    ret_df <-
+      switch(
+        type,
+        "2" = ,
+        II =
+          h_Anova_II_Chisq_mmrm(
+            mod,
+            vcov. = vcov.,
+            singular.ok = singular.ok,
+            ...
+          ),
+        "3" = ,
+        III =
+          h_Anova_III_Chisq_mmrm(
+            mod,
+            vcov. = vcov.,
+            singular.ok = singular.ok,
+            ...
+          )
+      )
+  }
   ret_df
 }
 
 
+
+
+
+#' Conduct Type-II or Type-III ANOVA with a Chi-squared Test Statistic.
+#'
+#' @inheritParams Anova.mmrm
+#'
+#' @returns `anova` object with one row per variable and columns `Chisq` (the
+#'   Chi-squared test statistic), `Df` (degrees of freedom), and `Pr(>Chisq)`
+#'   (p-value).
+#' @keywords internal
+h_Anova_II_Chisq_mmrm <- function(mod, vcov., singular.ok, ...) {
+  hypothesis_results <- function(term) {
+    which.term <- which(term == names)
```

Comment:
please use consistent `snake_case` for variables (even if this function might come from somewhere else originally)